src/main/java/uk/gov/pay/directdebit/payments/api/PaymentViewValidator.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.pay.directdebit.payments.api;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+public class PaymentViewValidator {
+
+    private static final Long MAX_PAGE_NUMBER = 500l;
+    private static final Long DEFAULT_PAGE_NUMBER = 1l;
+
+    public static Pair<Long, Long> validatePagination(Optional<Pair<Long, Long>> paginationParams) {
+        Long pageNumber = DEFAULT_PAGE_NUMBER;
+        Long displaySize = MAX_PAGE_NUMBER;
+        if (paginationParams.isPresent()) {
+            displaySize = paginationParams.get().getRight() > MAX_PAGE_NUMBER ? MAX_PAGE_NUMBER : paginationParams.get().getRight();
+            pageNumber = (paginationParams.get().getLeft() - 1) * displaySize;
+        }
+        return Pair.of(pageNumber, displaySize);
+    }
+
+    public Optional<List> validateQueryParams(List<Pair<String, Long>> nonNegativePairMap) {
```

Comment:
two problems here: `List` is not typed (ie. `List<String>`) and more generally `Optional<List>` does not make much sense. A list can be empty, and that's semantically equivalent to `Optional.empty()` as we use it, no need to add so much complexity. 
So, downstream, all you have to check is that the list is empty to make sure there are no errors.

Reply:
The validation is not fully done. It will be extended when I introduce to and from dates. See connector.